src/pages/Environments.js

```diff
@@ -130,7 +130,7 @@ const Environments = () => {
     const [newRuntimes, newDisks, galaxyDisks, newApps] = await Promise.all([
       Ajax(signal).Runtimes.list({ creator }),
       Ajax(signal).Disks.list({ creator }),
-      Ajax(signal).Disks.list({ creator, saturnApplication: 'galaxy' }),
+      Ajax(signal).Disks.list({ creator, saturnApplication: 'galaxy', includeLabels: 'saturnWorkspaceName' }),
```

Comment:
With my PR, we will just be making a single Disk.list call. See https://github.com/DataBiosphere/terra-ui/pull/2702/files#diff-db07f65dd0ba5edbd535b2a231709c1f5901f4e1c43f39c333165d97c1abacc4R133

Reply:
I don't think though that the Environments page needs the workspace name. It does not call `currentPersistentDisk`. The Environments page already partitions the disks by Google project, using disksByProject (the project property of the disk).

In general, why use saturnWorkspaceName instead of googleProject? With project per workspace, shouldn't they be the same thing?

Reply:
This PR is to fix a bug where in workspaces that were created before the project per workspace migration cause a Leo bug. Say a user has disk A in workspace A, deletes workspace A and then creates a new Galaxy in workspace B. Leo tries to attach the same disk from workspace A to the newly created Galaxy in workspace B. In Leo, two disks can't exist in the same workspace. So the fix is that there should be a new disk that gets created for workspace B which post PPW is the case for free

Reply:
Thanks @gpcarr, but I am still a bit confused about the issue. Looking at IA-2812, it seems that the issue is that 2 disks cannot exist in the same Google project. With pre-PPW workspaces, you could have 2 workspaces using the same project.

If that is the bug, why isn't it enough to just use googleProject? 

I am also not clear where the limitation is coming from (and therefore I'm unsure if this is something we need to be concerned about with Cromwell also). Can you clarify what the limitation is in Leo code itself?

Reply:
I also am not following this:

_"Say a user has disk A in workspace A, deletes workspace A and then creates a new Galaxy in workspace B. Leo tries to attach the same disk from workspace A to the newly created Galaxy in workspace B. In Leo, two disks can't exist in the same workspace. So the fix is that there should be a new disk that gets created for workspace B which post PPW is the case for free"_

This seems to be a case of a single disk being used in 2 workspaces (one of which has been deleted). Why is this a problem? How is it related to the Leo limitation that 2 disks can't exist in the same workspace (we have only 1 disk in play)?

Reply:
This is a Leo specific limitation. See [this line](https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala#L167) in Leo - we do not allow a disk to exist in two different workspaces. which is currently possible to do with pre-ppw migration workspaces. so we need to force the frontend to create a new disk if a user has 2 workspaces in the same project (these would be older ie pre-ppw workspaces)

Reply:
@gpcarr @kyuksel the Leo line of code linked above is not specific to Galaxy-- it is used by both Cromwell and Galaxy (see line 154). Therefore I think these changes should be made general for all apps, not Galaxy-specific. I have some questions about what a Custom app instance would do, but I don't think that is exposed to users, and I doubt the persistent disk logic is implemented for Custom. For simplicity, I think it would be best to make the restriction consistent for all apps until we have a use case where it would not be wanted.